sherpa/models/tests/test_parameter.py

```diff
@@ -25,200 +25,281 @@
+    # like it had passed.
+    #
+    with pytest.raises(ValueError):
+        p.frozen = arange(10)
+
+    afp = setUp_afp()
+    p.link = afp
+    assert p.frozen is True
+
+    p.link = None
+    p.freeze()
+    assert p.frozen is True
+
+    p.thaw()
+    assert p.frozen is False
+
+
+def test_link():
+    p = setUp_p()
+    assert p.link is None
+    assert p.val == 0.0
+
+    afp = setUp_afp()
+    afp.val = 7.3
+    p.link = afp
+    assert p.val == 7.3
+
+    p.unlink()
+    assert p.link is None
+
+    with pytest.raises(ParameterErr):
+        afp.link = p
+
+    with pytest.raises(ParameterErr):
+        p.link = 3
+
+    with pytest.raises(ParameterErr):
+        p.link = 3 * p + 2
+
+
+def test_iter():
+    p = setUp_p()
+    for part in p:
+        assert part is p
+
+
+def setUp_composite():
+    p = Parameter('model', 'p', 2)
+    p2 = Parameter('model', 'p2', 4)
+    return p, p2
+
+
+@pytest.mark.parametrize("op", (abs, operator.neg))
+def test_unop(op):
+    porig = setUp_p()
+    p = op(porig)
+    assert isinstance(p, UnaryOpParameter)
+    assert p.val == op(p.val)
+
+
+@pytest.mark.parametrize("op", [operator.add, operator.sub, operator.mul,
+                                operator.floordiv, operator.truediv, operator.mod,
+                                operator.pow])
+def test_binop(op):
+    p, p2 = setUp_composite()
+    for comb in (op(p, p2.val), op(p.val, p2),
+                 op(p, p2)):
+        assert isinstance(comb, BinaryOpParameter)
+        assert comb.val == op(p.val, p2.val)
+
+
+def test_iter_composite():
+    p, p2 = setUp_composite()
+    pnew = 3 * p + p2
+    parts = list(pnew)
+    assert len(parts) == 4
+    assert type(parts[0]) is BinaryOpParameter
+    assert type(parts[1]) is ConstantParameter
+    assert parts[2] is p
+    assert parts[3] is p2
+
+
+def test_complex_expression():
+    p, p2 = setUp_composite()
+    cmplx = (3 * p + p2) / (p ** 3.2)
+    p = p.val
+    p2 = p2.val
+    assert cmplx.val == (3 * p + p2) / (p ** 3.2)
+
+
+def setUp_link():
+    src1 = Gauss1D()
+    src2 = Gauss1D()
+    src1.pos.set(val=4, min=-10, max=10)
+    src2.pos.set(val=5, min=-20, max=20)
+    return src1, src2
+
+
+def tst_pos(gauss, pos, minval, maxval, frozen=False,
+            link=None):
+    assert gauss.val == pos
+    assert gauss.min == minval
+    assert gauss.max == maxval
+    assert gauss.frozen == frozen
+    if gauss.link is None or link is None:
+        assert gauss.link == link
+    else:
+        # These two asserts are just to point out that
+        # the values we check against here are different
+        # to the input values.
+        #
+        assert minval == -10
+        assert maxval == 10
+        tst_pos(gauss.link, pos, -20, 20)
+    assert gauss.default_val == pos
+
+    # Note: these are not minval/maxval but the defaults.
+    # How, where, and why are they set?
+    # assert gauss.default_min == minval
+    # assert gauss.default_max == maxval
```

Comment:
This makes no sense. Maybe it did at some point, but it does not now. I suggest to just remove those 4 lines of comments.

Reply:
I've taken out all the comments I could find.